src/Kiota.Builder/Refiners/CommonLanguageRefiner.cs

```diff
@@ -108,6 +108,28 @@ private static bool DoesAnyParentHaveAPropertyWithDefaultValue(CodeClass current
 
         return false;
     }
+    protected static void CorrectClassNames(CodeElement current, Func<string, string> refineClassName)
+    {
+        if (current is CodeClass currentClass &&
+            refineClassName(currentClass.Name) is string refinedClassName &&
+            !currentClass.Name.Equals(refinedClassName))
+        {
+            currentClass.Name = refinedClassName;
+        }
```

Comment:
How are we handling possible collisions in such cases? `animal_sound` and `animalSound` are going to resolve to the same class name. Can we consider having a flag during generation to enable such cases, i.e leave the responsibility to the generator to ensure that there are no collisions to the generator or have a fail-fast approach and validate that class names are not colliding before generation. This should apply to the property names as well. (the check should only be run if the flag to allow refined names is on)

Reply:
Spot on comment, we have been speaking about almost the same subject here:
https://github.com/microsoft/kiota/issues/2198#issuecomment-1406716523

My understanding is that we want to generate code that is going to be as idiomatic as possible in the target language.
At the same time, I would value a validation step that can catch those kinds of edge cases and fail the code generation instead of emitting incorrect code.

Reply:
@rkodev by flag do you mean a new command line argument? or something else?
We try to refrain from adding arguments for those sort of things to avoid confusion for the CLI users.